components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -1,284 +1,502 @@
+    };
 
-        let copyLink;
-        if (channel.type === Constants.OPEN_CHANNEL || channel.type === Constants.PRIVATE_CHANNEL) {
-            copyLink = (
-                <Menu.ItemAction
-                    id={`copyLink-${channel.id}`}
-                    onClick={this.copyLink}
-                    icon={<i className='icon-link-variant'/>}
-                    text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.copyLink', defaultMessage: 'Copy Link'})}
-                />
-            );
+    const toggleSubmenuDesktop = () => {
+        if (isMobile()) {
+            return;
         }
+        toggleSubmenu();
+    };
 
-        let addMembers;
-        if ((channel.type === Constants.PRIVATE_CHANNEL && this.props.managePrivateChannelMembers) || (channel.type === Constants.OPEN_CHANNEL && this.props.managePublicChannelMembers)) {
-            addMembers = (
-                <Menu.ItemAction
-                    id={`addMembers-${channel.id}`}
-                    onClick={this.addMembers}
-                    icon={<i className='icon-account-outline'/>}
-                    text={intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.addMembers', defaultMessage: 'Add Members'})}
-                />
-            );
+    const toggleSubmenuMobile = () => {
+        if (!isMobile()) {
+            return;
         }
+        toggleSubmenu();
+    };
+    const closeAllMenus = useCallback(() => {
+        setMenuVisible(false);
+        setSubMenuVisible(false);
+    }, []);
+
+    //****************** Open menu button ******************//
+    const menuTrigger = {
+        element:
+    <button
+        ref={buttonReference}
+        className={classNames(['SidebarMenu_menuButton'])}
+        onClick={toggleMenu}
+    >
+        <i className='icon-dots-vertical'/>
+    </button>,
+        ref: buttonReference,
+    };
 
-        let leaveChannelText = intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.leaveChannel', defaultMessage: 'Leave Channel'});
-        if (channel.type === Constants.DM_CHANNEL || channel.type === Constants.GM_CHANNEL) {
-            leaveChannelText = intl.formatMessage({id: 'sidebar_left.sidebar_channel_menu.leaveConversation', defaultMessage: 'Close Conversation'});
+    //****************** Open submenu menu item ******************//
+
+    const subMenuTrigger = {
+        element:
+    <MenuItem
+        id={`moveTo-${channel.id}`}
+        ref={menuItemReference}
+        label={intl.formatMessage({
+            id: 'sidebar_left.sidebar_channel_menu.moveTo',
+            defaultMessage: 'Move to...',
+        })}
+        leadingElement={
+            location === 'sidebar' ? (
+                <i className='icon-folder-move-outline'/>
+            ) : null
         }
+        onMouseEnter={toggleSubmenuDesktop}
+        onClick={toggleSubmenuMobile}
+        trailingElementLabel='selected'
+        trailingElement={<i className='icon-chevron-right'/>}
```

Comment:
also passing it in like this will cause a lot of unnecessary re-renders

and we can use the icon-component from `@mattermost/compass-icons/components` which makes aligning a lot easier